spring-cloud-alibaba-starters/spring-cloud-starter-alibaba-nacos-discovery/src/main/java/com/alibaba/cloud/nacos/discovery/NacosGatewayLocatorHeartBeat.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2013-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.cloud.nacos.discovery;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
+import com.alibaba.cloud.nacos.NacosDiscoveryProperties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.springframework.cloud.client.discovery.event.HeartbeatEvent;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.context.SmartLifecycle;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
+
+
+/**
+ * @author yuhuangbin
+ * @author ruansheng
+ */
+public class NacosGatewayLocatorHeartBeat implements ApplicationEventPublisherAware, SmartLifecycle {
+
+	private static final Logger log = LoggerFactory.getLogger(NacosGatewayLocatorHeartBeat.class);
+
+	private final NacosDiscoveryProperties nacosDiscoveryProperties;
+
+	private final ThreadPoolTaskScheduler taskScheduler;
+	private final AtomicLong nacosWatchIndex = new AtomicLong(0);
+	private final AtomicBoolean running = new AtomicBoolean(false);
+	private ApplicationEventPublisher publisher;
+	private ScheduledFuture<?> watchFuture;
+
+	public NacosGatewayLocatorHeartBeat(NacosDiscoveryProperties nacosDiscoveryProperties) {
+		this.nacosDiscoveryProperties = nacosDiscoveryProperties;
+		this.taskScheduler = getTaskScheduler();
+	}
+
+	private static ThreadPoolTaskScheduler getTaskScheduler() {
+		ThreadPoolTaskScheduler taskScheduler = new ThreadPoolTaskScheduler();
+		taskScheduler.setBeanName("Nacos-SCG-HeartBeat-Task-Scheduler");
```

Comment:
Please check the accuracy whether just used by SCG, such as Zuul?

Reply:
Just used by SCG , `HeartBeatEvent` is invalid for zuul ,  zuul using `EnvironmentChangeEvent` to refresh route list , see `SimpleHostRoutingFilter#onPropertyChange` ( spring-cloud-netflix-zuul )

Reply:
> EnvironmentChangeEvent

I found it also uses Heartbeat to refresh route in Zuul.
![image](https://user-images.githubusercontent.com/25423978/199415644-67bbebde-3d0d-4ae0-9cd6-b830591c3e42.png)

Reply:
DiscoveryClientRouteLocator enabled by @EnableZuulProxy , No property can disable `DiscoveryClientRouteLocator` , we can only import `spring-cloud-netflix-zuul` module with `optional` and using `@ConditionalOnBean(Marker.class)` to enable `GatewayLocatorHeartBeatPublisher`

Reply:
`Nacos-SCG-HeartBeat-Task-Scheduler` it is recommended to modify to `HeartBeat-Task-Scheduler` is better?

Reply:
> `Nacos-SCG-HeartBeat-Task-Scheduler` it is recommended to modify to `HeartBeat-Task-Scheduler` is better?

sounds good

Reply:
> Please check the accuracy whether just used by SCG, such as Zuul?

2021.x no longer supported zuul , supported in 2.2.x